test/e2e/declarative_test_fwk/model.py

```diff
@@ -53,11 +53,14 @@ class DeleteDict(ResourceDict):
     pass
 
 
-# fields for 'expect' field in a test step
-class ExpectDict(TypedDict, total=False):
+# fields for 'expect_k8s' field in a test step
+class ExpectK8SDict(TypedDict, total=False):
     spec: Dict
     status: Dict
 
+#fields for 'expect_aws' field in a test step
+class ExpectAPIResponseDict(TypedDict, total=False):
+    expect_aws: Dict
```

Comment:
This is going to lead to a YAML that looks like:
```yaml
...
expect_aws:
  expect_aws:
    ...
```

Which isn't necessary. I don't think you need to define a type `ExpectAPIResponseDict`. Just, only line 74, do `expect_aws: Dict`. It doesn't need a wrapper class.

Reply:
Updated to regular dict